File: mcp-server-typescript-express/src/tools/hello.ts
import type { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
import { z } from "zod";

export function registerHelloTool(server: McpServer) {
  server.tool(
    "hello",
    "Returns a friendly greeting. Useful as a smoke test that the server is reachable and your Keycard policy allows tool calls.",
    { name: z.string().min(1).describe("Who to greet") },
    async ({ name }) => ({
      content: [
        {
          type: "text",
          text: `Hello, ${name}!`,
        },
      ],
    }),
  );
}

File: mcp-server-python-mcp/tools/hello.py
from mcp.server.mcpserver import MCPServer


def register_hello_tool(mcp: MCPServer) -> None:
    @mcp.tool()
    def hello_world(name: str) -> str:
        """Say hello to an authenticated user."""
        return f"Hello, {name}! You are authenticated."
